File: Smart_Booking_Backend/src/main/java/com/smartbooking/service/BookmarkService.java
package com.smartbooking.service;

import com.smartbooking.entity.Bookmark;
import java.util.List;
import java.util.Optional;

public interface BookmarkService {
    List<Bookmark> getAllBookmarks();
    Optional<Bookmark> getBookmarkById(Long id);
    Bookmark createBookmark(Bookmark bookmark);
    Bookmark updateBookmark(Long id, Bookmark bookmarkDetails);
    void deleteBookmark(Long id);
}

File: smart_bookmark_frontend/src/services/BookmarkService.js
import axios from 'axios';

const API_URL = "http://localhost:8080/api/bookmarks";

class BookmarkService {
    
    
    getAllBookmarks() {
        return axios.get(API_URL);
    }

    
    createBookmark(bookmark) {
        return axios.post(API_URL, bookmark);
    }

    
    getBookmarkById(bookmarkId) {
        return axios.get(API_URL + '/' + bookmarkId);
    }

    
    updateBookmark(bookmarkId, bookmark) {
        return axios.put(API_URL + '/' + bookmarkId, bookmark);
    }

    
    deleteBookmark(bookmarkId) {
        return axios.delete(API_URL + '/' + bookmarkId);
    }
}

export default new BookmarkService();